### DISCUSSION TITLE
problem log off

### DISCUSSION THREAD

**Discussion by gogi100** [rejetto/hfs#171] [General]
a user is loged and redirected on next page 
![log on](https://user-images.githubusercontent.com/21057856/226904263-1e936180-ae93-4a96-a55c-d82d0cd67ce2.jpg)

if user clicks on X in web browser
![x windows](https://user-images.githubusercontent.com/21057856/226904921-9a4d163a-8653-4be5-92c9-3721f4531838.jpg)

then the users again backs to address http://192.168.99.247, he does not receive dialog login than 
![back](https://user-images.githubusercontent.com/21057856/226905349-2b8c7f0a-e6ba-425d-bcfa-3468d014f01c.jpg)

the session is stayed active. how i break session if the user clicks on X button, near log off button that exists in web application?

**Comment by rejetto** [comment]
There is no such option at the moment

**Reply by gogi100** [reply]
is there a possibility that when the user clicks on x, he will be returned to the login dialog

**Comment by rejetto** [comment]
i made some tests, and I've found no consistent way to get this result.

Apparently there is an official way, but is very unreliable. Mozilla states
> Session cookies are deleted when the current session ends. The browser defines when the "current session" ends, and some browsers use session restoring when restarting. This can cause session cookies to last indefinitely.

Just tested with Chrome on mac, and having the option to restore tabs after restarting is enough to NOT delete these cookies.

So, I tried to try to ask (the server) for logout as soon as the user leaves the page. There's not enough time, unless I demand the browser to ask user for confirmation. This way the logout is done while the user looks at the dialog. The problem is that the user may not accept the dialog, stays on HFS, but now he is logged out.
Another problem is that this is triggered not only when user click X, but also if user reloads, or change website.

So, if you still want to do it, just add this in custom html "top", but it will work only starting with version 0.44.0
```html
<script>
addEventListener('beforeunload', (e) => HFS.logout() && e.returnValue = 'bye!')
</script>
```

I can provide as usual a preview version if you want

**Comment by rejetto** [comment]
0.44 beta has a feature that may be interesting for you: "keep session alive", on by default, can be disabled
https://github.com/rejetto/hfs/discussions/221

also, check the SESSION_DURATION
https://github.com/rejetto/hfs#hidden-features
